File: Wolf/Wolf.cpp
#include "Wolf.h"

Wolf::Wolf(World *world, int x, int y, int parent1Id, int parent2Id) :
  Animal(WOLF_STRENGTH, WOLF_INITIATIVE, WOLF_SYMBOL, WOLF_COLOR, world, x, y, parent1Id, parent2Id) {}

void Wolf::action() {
  Animal::action();
}

void Wolf::collision(Organism *organism) {
  Animal::collision(organism);
}

Organism* Wolf::createChild(int x, int y, int parent1Id, int parent2Id) {
  return new Wolf(this->getWorld(), x, y, parent1Id, parent2Id);
}

std::string Wolf::getType() const { return "Wolf"; }

File: src/Wolf/Wolf.java
package Wolf;

import Animal.Animal;
import Field.Field;
import Organism.*;
import Settings.Settings;
import World.World;

public class Wolf extends Animal {

    public Wolf(World world, Field field, int parent1Id, int parent2Id) {
        super(new OrganismData(world.getNextOrganismId(), Settings.WOLF_INITIATIVE, Settings.WOLF_STRENGTH, Settings.WOLF_IMAGE, parent1Id, parent2Id), world, field);
    }

    @Override
    public String getType() {
        return "Wolf";
    }

    @Override
    public void reproduce(Field field, int parent1Id, int parent2Id) {
        Organism wolf = new Wolf(this.world, field, parent1Id, parent2Id);
        this.world.addOrganism(wolf);
    }
}
